dataproc/submit_job_to_cluster.py

```diff
@@ -42,222 +60,229 @@ def get_region_from_zone(zone):
+    to `bucket_name."""
     job_details = {
-        'projectId': project,
-        'job': {
-            'placement': {
-                'clusterName': cluster_name
-            },
-            'pysparkJob': {
-                'mainPythonFileUri': 'gs://{}/{}'.format(bucket_name, filename)
-            }
+        'placement': {
+            'cluster_name': cluster_name
+        },
+        'pyspark_job': {
+            'main_python_file_uri': 'gs://{}/{}'.format(bucket_name, filename)
         }
     }
-    result = dataproc.projects().regions().jobs().submit(
-        projectId=project,
-        region=region,
-        body=job_details).execute()
-    job_id = result['reference']['jobId']
-    print('Submitted job ID {}'.format(job_id))
+
+    result = dataproc.submit_job(
+        project_id=project, region=region, job=job_details)
+    job_id = result.reference.job_id
+    print('Submitted job ID {}.'.format(job_id))
     return job_id
 # [END dataproc_submit_pyspark_job]
 
 
 # [START dataproc_delete]
 def delete_cluster(dataproc, project, region, cluster):
-    print('Tearing down cluster')
-    result = dataproc.projects().regions().clusters().delete(
-        projectId=project,
-        region=region,
-        clusterName=cluster).execute()
+    """Delete the cluster."""
+    print('Tearing down cluster.')
+    result = dataproc.delete_cluster(
+        project_id=project, region=region, cluster_name=cluster)
     return result
 # [END dataproc_delete]
 
 
 # [START dataproc_wait]
 def wait_for_job(dataproc, project, region, job_id):
+    """Wait for job to complete or error out."""
     print('Waiting for job to finish...')
     while True:
-        result = dataproc.projects().regions().jobs().get(
-            projectId=project,
-            region=region,
-            jobId=job_id).execute()
+        job = dataproc.get_job(project, region, job_id)
         # Handle exceptions
-        if result['status']['state'] == 'ERROR':
-            raise Exception(result['status']['details'])
-        elif result['status']['state'] == 'DONE':
+        if job.status.State.Name(job.status.state) == 'ERROR':
+            raise Exception(job.status.details)
+        elif job.status.State.Name(job.status.state) == 'DONE':
             print('Job finished.')
-            return result
+            return job
 # [END dataproc_wait]
 
 
-# [START dataproc_get_client]
-def get_client():
-    """Builds an http client authenticated with the service account
-    credentials."""
-    dataproc = googleapiclient.discovery.build('dataproc', 'v1')
-    return dataproc
-# [END dataproc_get_client]
-
+def main(project_id,
+         zone,
+         cluster_name,
+         bucket_name,
+         pyspark_file=None,
+         create_new_cluster=True,
+         global_region=True):
+
+    # [START dataproc_get_client]
+    if global_region:
+        region = 'global'
+        # Use the default gRPC global endpoints.
+        client_transport = ''
```

Comment:
Usually we use None for unspecified values, but this could work if the client checks for falsiness rather than explicitly checking for None.

Reply:
I will push new file versions that eliminate the need for "" or None.